python/dgl/graph.py

```diff
@@ -1165,6 +1170,10 @@ def from_networkx(self, nx_graph, node_attrs=None, edge_attrs=None):
                 [2., 2., 2., 2.],
                 [1., 1., 1., 1.]])
         """
+        nx_graph = nx_graph.to_directed()
+        # Relabel nodes using consecutive integers
+        nx_graph = nx.convert_node_labels_to_integers(nx_graph)
```

Comment:
Not needed. The graph index already calls it (see below).

Reply:
When we convert networkx graph into dgl graph with node features, it assumes that nodes are labeled with consecutive integers due to `for nid in range(self.number_of_nodes()):` in line 1193. Initially I thought relabeling the graph will help circumvent not only this issue for node feature but also for edge feature. It turns out that we cannot circumvent line 1208 anyway and I guess you are right that we should remove line 1173 and change line 1193 to be `for nid in nx_graph.nodes()` instead.

Reply:
> It turns out that we cannot circumvent line 1208 anyway

Why is that? Also, technically `for nid in nx_graph.nodes()` does not assume any order so `for nid in range(self.number_of_nodes())` is clearer.

Reply:
1. @ylfdq1118 Could you please take a look at whether my understanding is correct for the rest of item 1. From [L1196-1199 of graph.py](https://github.com/dmlc/dgl/blob/master/python/dgl/graph.py#L1196), we assume that there exists a prior specification of edge ids and the edge attributes are stored in the rows specified by these edge ids. @jermainewang Then if the edge ids are broken after we remove some nodes there will still be a problem. I guess this is not something that we need to take care.
2. If you do `for nid in range(self.number_of_nodes())`, since we did not relabel the nodes with consecutive integers in the `from_networkx` in `graph.py`, a KeyError can occur. I can do `for nid in sorted(nx_graph.nodes())` if this works for you.

Reply:
I see what happens here. Then, how about this?
(1) Relabel the networkx in `graph.py` because the order is required to set node features.
(2) `graph_index.py` will not relabel the nx graph and assume it is correct.

The benefit is we don't need to sort the nodes twice. As long as the `graph_index.from_networkx` is only called by `graph.from_networkx`, it would be fine. What do you think?

Reply:
Sounds good to me. Done.

Reply:
For the edge id, maybe we could loop over the list and check whether they is a `None`? and if so, we raise error about the consecutive edge id blah blah.

Reply:
Yeah I can do that. `attrs['id']` can also lead to an index out of range error, shall we also handle that?

Reply:
Yeah, that's a good idea.

Reply:
Done.